### PR TITLE
Disable editing key port for old servers

### PR DESCRIPTION
**Pull request by JonathanDCohen**
tested by running a local shadowbox server from the daily-2019-07-30 image on quay.io.

### CODE REVIEW

File: src/server_manager/ui_components/outline-server-settings.html [hunk 1]
```diff
@@ -202,6 +202,7 @@ <h3>[[localize('settings-metrics-header')]]</h3>
         serverHostname: {type: String, value: null},
         serverManagementApiUrl: {type: String, value: null},
         serverPortForNewAccessKeys: {type: Number, value: null},
+        isAccessKeyPortEditable: Boolean,
```
Comment (alalamav) [inline src/server_manager/ui_components/outline-server-settings.html]: Initalize explicitly: `{type: Bolean, value: false}`. Same for the one in server.

File: src/server_manager/ui_components/outline-server-settings.html [hunk 2]
```diff
@@ -253,6 +254,7 @@ <h3>[[localize('settings-metrics-header')]]</h3>
         this.$.keysPortCancelButton.disabled = !enabled;
       },
       _handleKeysPortClick: function() {
+        if (!this.isAccessKeyPortEditable) { return; }
```
Comment (alalamav) [inline src/server_manager/ui_components/outline-server-settings.html]: nit: maybe drop the return to the next line

File: src/server_manager/web_app/app.ts [hunk 3]
```diff
@@ -53,6 +53,15 @@ function isManualServer(testServer: server.Server): testServer is server.ManualS
   return !!(testServer as server.ManualServer).forget;
 }
 
+async function isAccessKeyPortEditable(server: server.Server): Promise<boolean> {
+  try {
+    await server.setPortForNewAccessKeys(server.getPortForNewAccessKeys());
```
Comment (alalamav) [inline src/server_manager/web_app/app.ts]: As discussed offline, let's use a negative value as a sentinel to guarantee idempotence.

Comment (JonathanDCohen) [reply src/server_manager/web_app/app.ts]: Done

File: src/server_manager/ui_components/outline-server-settings.html (hunk 2 above)
Comment (JonathanDCohen) [reply src/server_manager/ui_components/outline-server-settings.html]: Done

File: src/server_manager/ui_components/outline-server-settings.html (hunk 1 above)
Comment (JonathanDCohen) [reply src/server_manager/ui_components/outline-server-settings.html]: Done

File: src/server_manager/web_app/app.ts [hunk 4]
```diff
@@ -53,6 +53,17 @@ function isManualServer(testServer: server.Server): testServer is server.ManualS
...
 }
 
+async function isAccessKeyPortEditable(server: server.Server): Promise<boolean> {
+  try {
+    // -1 always fails, making this idempotent
+    await server.setPortForNewAccessKeys(-1);
+    // unreachable unless something went HORRIBLY WRONG
+    return false;
+  } catch (e) {
+    return !e.isNetworkError() && e.response.code !== 404;
```
Comment (alalamav) [inline src/server_manager/web_app/app.ts]: I don't see `code` in the `Response`interface; I think it is `status`? Also, we may want to check `e.response.code === 400` (we may get 5XXs)?

Comment (JonathanDCohen) [reply src/server_manager/web_app/app.ts]: Done

**Pull request merged**